lib/lockbox/encryptor.rb

```diff
@@ -39,7 +39,20 @@ def decrypt(ciphertext, **options)
           error_classes << RbNaCl::LengthError if defined?(RbNaCl::LengthError)
           error_classes << RbNaCl::CryptoError if defined?(RbNaCl::CryptoError)
           if error_classes.any? { |ec| e.is_a?(ec) }
-            raise DecryptionError, "Decryption failed" if i == @boxes.size - 1
+            if i == @boxes.size - 1
+              # See https://withpersona.slack.com/archives/C02TE6J9JB0/p1716411555974969
+              # When possible, we re-encrypt the ciphertext, just in case whatever was passed to lockbox
+              # was actually plaintext somehow. Then, we log the ciphertext and the error class,
+              # so we can investigate further.
+              begin
+                re_encrypted_ciphertext = box.encrypt(ciphertext, **options)
+                encoded_payload = Base64.strict_encode64(re_encrypted_ciphertext)
+                raise DecryptionError, "Decryption failed. Error: #{e.class}. Encrypted payload: #{encoded_payload}"
+              rescue ArgumentError
```

Comment:
Is `ArgumentError` the only thing we should be rescuing here? If not, we might want something more general, like:

```
encoded_payload = begin
  re_encrypted_ciphertext = box.encrypt(ciphertext, **options)
  Base64.strict_encode64(re_encrypted_ciphertext)
rescue StandardError
  encoded_payload = Base64.strict_encode64(ciphertext)
  raise DecryptionError, "Decryption failed. Error: #{e.class}. Couldn't encrypt payload. Original payload: #{encoded_payload}"
end
raise DecryptionError, "Decryption failed. Error: #{e.class}. Encrypted payload: #{encoded_payload}"
```

Reply:
I suppose we _could_ rescue something more general here, but the reason I rescue `ArgumentError` is that's what we'd hit if the `box` didn't have the ability to encrypt.

Reply:
I'll add a comment explaining why I only rescue `ArgumentError`. Re-reading the code, I agree it's not clear why I'm rescuing only that.

Reply:
I wasn't really too worried anyway. Even if it had missed an actual exception that could happen, the worst case is no worse than previous: we'd still get an exception but just not have any of the new information (just as before). So feel free to merge either way.

Reply:
I just pushed up a change to clean this up a bit.